src/factory/cosmosService.ts

```diff
@@ -0,0 +1,77 @@
+import { Container, CosmosClient } from "@azure/cosmos";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import * as TE from "fp-ts/TaskEither";
+import { Either } from "fp-ts/lib/Either";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import {
+  getChangeFeedIteratorOptions,
+  processChangeFeed,
+} from "../capturer/cosmos/cosmos";
+import {
+  cosmosConnect,
+  getContainer,
+  getDatabase,
+  getItemById,
+} from "../capturer/cosmos/utils";
+import {
+  CDCService,
+  DBClient,
+  DatabaseConfig,
+  DatabaseService,
+} from "./service";
+
+export const cosmosDBService: DatabaseService = {
+  getDatabase,
+  getResource: getContainer,
+  connect: (config: DatabaseConfig): Either<Error, DBClient> =>
+    cosmosConnect(config.connection, config.connection),
+};
+
+export const cosmosCDCService: CDCService = {
+  processChangeFeed: (
+    client: CosmosClient,
+    database: string,
+    resource: string,
+    leaseResource?: string,
+    prefix?: string
+  ): TaskEither<Error, void> =>
+    pipe(
+      E.Do,
+      E.bind("database", () => cosmosDBService.getDatabase(client, database)),
+      E.bind("container", ({ database }) =>
+        pipe(
+          cosmosDBService.getResource(database, resource),
+          E.map((c) => c as Container)
```

Comment:
do we need explicit cast?

Reply:
yes because getResource returns a Resource that is : 
` export type Resource = Container | Collection;
`
in the method we need a Container, so it seems to be necessary.
Could you suggest an alternative ?